tests/api/api.spec.ts

```diff
@@ -137,6 +137,72 @@ suite("@dataform/api", () => {
         expect(action).to.include({ type: "assertion" });
       });
     });
+
+    suite("pre and post ops", () => {
+      for (const warehouse of [
+        "bigquery",
+        "postgres",
+        "redshift",
+        "sqldatawarehouse",
+        "snowflake"
+      ]) {
+        const graph: dataform.ICompiledGraph = dataform.CompiledGraph.create({
+          projectConfig: { warehouse: "redshift" },
+          tables: [
+            {
+              name: "a",
+              target: { schema: "schema", name: "a" },
+              type: "incremental",
+              query: "foo",
+              incrementalQuery: "incremental foo",
+              preOps: ["preOp"],
+              incrementalPreOps: ["incremental preOp"],
+              postOps: ["postOp"],
+              incrementalPostOps: ["incremental postOp"]
+            }
+          ],
+          dataformCoreVersion: "1.4.9"
+        });
+
+        test(`${warehouse} when running non incrementally`, () => {
+          const action = new Builder(graph, {}, TEST_STATE).build().actions[0];
+          expect(action.tasks[0]).eql(
```

Comment:
Would be a slightly tighter test if you just checked array equality on all the tasks, including the query (e.g. this would not catch our previous issue of duplicated pre / post ops)

Reply:
discussed on slack but for posterity: it's a fair point, I'm trying to avoid having per-warehouse SQL in here though which would be hard to avoid in that case.

perhaps we should just give up and have per-warehouse DB adapter tests.